### PR TITLE
[clang-tidy][NFC] Replace usages of `DeclSpec::TQ` with `Qualifiers::TQ`

### PR DESCRIPTION
**Pull request by Endilll**
This patch improves, but doens't fully resolve the layering violation, which stems from relying on Sema. There's one function that needs to convert enumerator to a string (`buildQualifier` in `FixItHintUtils.cpp`), but `Qualifiers::TQ` doesn't offer such function. Even more, the set of enumerators is not complete compared to `DeclSpec::TQ`, so I'm afraid that this would be a functional change.

### CODE REVIEW

File: clang-tools-extra/clang-tidy/utils/FixItHintUtils.cpp
Comment (cor3ntin) [inline clang-tools-extra/clang-tidy/utils/FixItHintUtils.cpp:79]: I feel like this is moving the problem to the source file. Can we use `Qualifiers::getAsString()` and remove the `DeclSpec` dependence entirely?

Comment (Endilll) [reply clang-tools-extra/clang-tidy/utils/FixItHintUtils.cpp:79]: > Qualifiers::getAsString()

From that I saw, it does much more than `getSpecifierName`. It also can serialize only a subset of qualifiers, because `Qualifiers::TQ` doesn't seem to support `__unaligned` and `_Atomic`. I think what I've done is an improvement over status quo, and I'd like to avoid going into the rabbit hole.

Comment (cor3ntin) [reply clang-tools-extra/clang-tidy/utils/FixItHintUtils.cpp:79]: In practice, this function only ever to print "const", afaik (and nothing else).

You could even reimplement it inline

```cpp
const char getSpecifierName(Qualifier T) {
  switch (T) {
     case Qualifiers::Qualifier:   return "const";
     default:
          assert("not supported");
  }
  return "";
}
```

But really if `Qualifiers::getAsString` passes the test, you should just use that.

Comment (Endilll) [reply clang-tools-extra/clang-tidy/utils/FixItHintUtils.cpp]: I updated the PR. Is it good now?

Comment (cor3ntin) [reply clang-tools-extra/clang-tidy/utils/FixItHintUtils.cpp]: LGTM.

**Comment by cor3ntin** [comment]
LGTM, Thanks - However this is a clang-tidy change so you should wait for its maintainers to approve it

**Comment by Endilll** [comment]
> However this is a clang-tidy change so you should wait for its maintainers to approve it

It would be nice to know who I'm waiting for, because there are not listed maintainers for clang-tidy. Is it Aaron then?

**Review by HerrCai0907 — APPROVED** [review]
LGTM. thanks!

**Pull request merged**
